### DISCUSSION TITLE
aiojobs : how to cleanup closed jobs? Memory leak expected

### DISCUSSION THREAD

**Discussion by averatio** [aio-libs/aiohttp#7657] [General]
Hello
I use aiojobs in aiohttp, It works perfect. A web handler spaws a job at each http call, so thousands of jobs are created every day.
My understanding is that a closed job is still an object in memory which is referenced by the scheduler, so not garbaged.
Since the scheduler is created at aiohttp server level and not at handler level, it is never garbaged, so I expect a memory leak of all closed jobs.
Are the previous statements correct?
How to avoid a memory leak?
Regards
AVE

**Comment by Dreamsorcerer** [comment]
First, should probably be asking on the aiojobs repo, this has nothing to do with aiohttp.
Second, the scheduler only holds references to running jobs. So, as long as you are not holding any references yourself, then it will be garbage collected and behave as expected.
